src/GitVersionCore.Tests/StringFormatWithExtensionTests.cs

```diff
@@ -0,0 +1,75 @@
+﻿using System;
+
+using GitVersion;
+using NUnit.Framework;
+
+namespace GitVersionCore.Tests
+{
+    [TestFixture]
+
+    public class StringFormatWithExtensionTests
+    {
+        [Test]
+        public void FormatWith_NoTokens()
+        {
+            var propertyObject = new { };
+            var target = "Some String without tokens";
+            var expected = target;
+            var actual = target.FormatWith(propertyObject);
+            Assert.AreEqual(expected, actual);
+        }
+
+        [Test]
+        public void FormatWith_SingleSimpleToken()
+        {
+            var propertyObject = new { SomeProperty = "SomeValue" };
+            var target = "{SomeProperty}";
+            var expected = "SomeValue";
+            var actual = target.FormatWith(propertyObject);
+            Assert.AreEqual(expected, actual);
+        }
+
+        [Test]
+        public void FormatWith_MultipleTokensAndVerbatimText()
+        {
+            var propertyObject = new { SomeProperty = "SomeValue", AnotherProperty = "Other Value" };
+            var target = "{SomeProperty} some text {AnotherProperty}";
```

Comment:
Will these "bare" format strings work for the variables provided by GitVersion? Perhaps we could make that explicit with a `gv:` prefix?

Reply:
This testcase just tests the existing functionality and specifically tests the `FormatWith` method. I guess this should not be a concern

Reply:
Note that introducing a "gv:"-prefix for git version properties would break existing config files. It may be acceptable to introduce breaking changes while still in beta phase, but that is something I'd rather do in a dedicated PR.

Reply:
True, scrap that. No prefix means GitVersion's own variables.